src/pages/workspaces/workspace/Data.js

```diff
@@ -417,32 +422,20 @@ const WorkspaceData = _.flow(
   const signal = useCancellation()
   const asyncImportJobs = useStore(asyncImportJobStore)
 
-
-  // Helpers
-  const getSelectionType = () => {
-    const referenceData = getReferenceData(attributes)
-    return Utils.cond(
-      [!selectedDataType, () => 'none'],
-      [selectedDataType === localVariables, () => 'localVariables'],
-      [selectedDataType === bucketObjects, () => 'bucketObjects'],
-      [_.isArray(selectedDataType), () => 'snapshots'],
-      [_.includes(selectedDataType, _.keys(referenceData)), () => 'referenceData'],
-      () => 'entities'
-    )
-  }
-
```

Comment:
So basically you obviated the need for this entire conditional by choosing a more appropriate data structure to hold the state. That's pretty great.